cpp/src/parquet/encoding.cc

```diff
@@ -3370,20 +3375,23 @@ class DeltaByteArrayDecoderImpl : public DecoderImpl, virtual public TypedDecode
         throw ParquetException("excess expansion in DELTA_BYTE_ARRAY");
       }
     }
-    PARQUET_THROW_NOT_OK(buffered_data_->Resize(data_size));
+    PARQUET_THROW_NOT_OK(buffered_data_->Resize(data_size, false));
```

Comment:
```suggestion
    PARQUET_THROW_NOT_OK(buffered_data_->Resize(data_size, /*shrink_to_fit=*/false));
```

Reply:
BTW, we lose the chance to release bulk memory if shrink_to_fit is false. Changing the default behavior of `buffered_prefix_length_` above may not make big difference but the actual size of `buffered_data_` may vary a lot during the decoding.

Reply:
Let me leave it first, maybe we need a better method to release memory?